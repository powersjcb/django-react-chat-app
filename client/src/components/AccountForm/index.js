import React from 'react'
import PropTypes from "prop-types"
import { Link } from 'react-router-dom'
import { Input, Label, Box } from 'rebass'


class AccountForm extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      username: '',
      password: '',
    }
  }

  render() {
    if (this.props.loggedIn) {
      return null
    }
    return (
      <Box width={1/2} m={"auto"}>
        <form onSubmit={submitEvent => {
          submitEvent.preventDefault()
          this.props.submitForm(
            this.state.username,
            this.state.password)
        }}>
          <h3>{this.props.formLabel}</h3>

          <Label>
            Username
            <Input
              type="text"
              onChange={e => this.setState({username: e.target.value})}
            />
          </Label>

          <Label>
            Password
            <Input
              type="password"
              onChange={e => this.setState({password: e.target.value})}
            />
          </Label>

          <Input type="submit" value={this.props.submitLabel}/>
        </form>

        {!this.props.signup && (
            <Link to="/signup">Signup for an account</Link>
        )}
      </Box>
    );
  }

}

AccountForm.propTypes = {
  formLabel: PropTypes.string,
  submitLabel: PropTypes.string,
  submitForm: PropTypes.func,
  loggedIn: PropTypes.bool,
  signup: PropTypes.bool,
}

export default AccountForm
