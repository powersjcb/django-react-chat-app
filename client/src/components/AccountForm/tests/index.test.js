import React from 'react'
import AccountForm from '../index'
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

test('Renders login page', () => {
  const component = renderer.create(
    <MemoryRouter>
      <AccountForm />
    </MemoryRouter>
  )

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

})