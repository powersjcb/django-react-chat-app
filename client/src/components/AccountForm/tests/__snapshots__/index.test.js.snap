// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders login page 1`] = `
<div
  className="sc-bdVaJa iFgAQR"
>
  <form
    onSubmit={[Function]}
  >
    <h3 />
    <label
      className="sc-kgoBCf gNzOsm"
    >
      Username
      <input
        className="sc-kGXeez cwsHfV"
        onChange={[Function]}
        type="text"
      />
    </label>
    <label
      className="sc-kgoBCf gNzOsm"
    >
      Password
      <input
        className="sc-kGXeez cwsHfV"
        onChange={[Function]}
        type="password"
      />
    </label>
    <input
      className="sc-kGXeez cwsHfV"
      type="submit"
      value={undefined}
    />
  </form>
  <a
    href="/signup"
    onClick={[Function]}
  >
    Signup for an account
  </a>
</div>
`;
